src/Search.java

```diff
@@ -0,0 +1,90 @@
+import java.util.InputMismatchException;
+import java.util.Scanner;
+
+public class Search {
+	static Scanner scanner=new Scanner(System.in);
+	public static void main(String[] args)
+	{
+		System.out.println("Enter the size of array");
+		int temp=getInput();
+		if(temp<=0)
+		{
+			scanner.nextLine();
+			temp=getInput();
+		}
+		int arr[]=new int[temp];
+		for(int loop_var=0;loop_var<temp;loop_var++)
+		{
+			System.out.print("Enter element "+(loop_var+1)+" = ");
+			arr[loop_var]=getInput();
+		}
+		System.out.println("Enter the element to search");
+		int to_search=getInput();
+		Search search=new Search();
+		System.out.println("Index of "+to_search+" = "+search.binarySearch(arr,0,arr.length,to_search)); //It will only work for sorted array
+		System.out.println("Index of "+to_search+" = "+search.linearSearch(arr, to_search));
+	}
+	/**
+	 * @return numeric value only
+	 */
+	static int getInput()
+	{
+		int temp = 0;
+		try
+		{
+			temp=scanner.nextInt();
+			return temp;
+		}
+		catch(InputMismatchException e)
+		{
+			System.out.println("Enter numeric value only");
+			scanner.nextLine();
+			return getInput();
+		}
+	}
+	/**
+	 * @param arr array to be searched
+	 * @param value_to_search value to be searched
+	 * @param left_index starting position for binary search
+	 * @param right_index ending position
+	 * @return index of search value if found in array else -1
+	 */
+	int binarySearch(int arr[], int left_index, int right_index, int value_to_search)
```

Comment:
You should also provide an overloaded method for binary search, which only have a search kay and an array as method parameter.

Reply:
done